pkg/netconfig/netconfig.go

```diff
@@ -0,0 +1,366 @@
+		err = DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("4", "-t", "mangle", "-A", "POSTROUTING", "-o", dummyDeviceName, "-j", ipTablesLoggingChain)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Error creating ip%stables rule for logging packets to dummy device: %v\n", "", err))
+		}
+	}
+
+	if err := DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("6", "-t", "mangle", "-C", "POSTROUTING", "-o", dummyDeviceName, "-j", ipTablesLoggingChain); err != nil {
+		err = DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("6", "-t", "mangle", "-A", "POSTROUTING", "-o", dummyDeviceName, "-j", ipTablesLoggingChain)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Error creating ip%stables rule for logging packets to dummy device: %v\n", "6", err))
+		}
+	}
+	fmt.Println("Created iptables rules for logging packets to dummy device.")
+	return nil
+}
+
+func GetBlackholeRoutes(ipVersion string) ([]string, error) {
+	blackholeRoutes := []string{}
+
+	ipOut, err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand(ipVersion, "route")
+
+	if err != nil {
+		return blackholeRoutes, err
+	}
+
+	lines := strings.Split(ipOut.String(), "\n")
+	for _, line := range lines {
+		if strings.Contains(line, dummyDeviceName) && !strings.Contains(line, "fe80::/64") {
+			fields := strings.Fields(line)
+			addr := fields[0]
+			blackholeRoutes = append(blackholeRoutes, addr)
+		}
+	}
+
+	return blackholeRoutes, nil
+}
+
+func DeleteRoutes(ipVersion string, addrs []string) error {
+	if len(addrs) == 0 {
+		return nil
+	}
+	fmt.Printf("Deleting %d routes.\n", len(addrs))
+	for i, addr := range addrs {
+		addrs[i] = "route del " + addr + " dev dummy0"
+	}
+	err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteBatchCommand(ipVersion, strings.Join(addrs, "\n"))
+
+	if err != nil {
+		return fmt.Errorf("Error deleting blackhole routes: %v\n", err)
+	}
+	fmt.Printf("Deleted %d routes.\n", len(addrs))
+	return nil
+}
+
+func AddRoutes(ipVersion string, addrs []string) error {
+	if len(addrs) == 0 {
+		return nil
+	}
+	fmt.Printf("Adding %d routes.\n", len(addrs))
+	for i, addr := range addrs {
+		addrs[i] = "route add " + addr + " dev dummy0"
+	}
+	err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteBatchCommand(ipVersion, strings.Join(addrs, "\n"))
+
+	if err != nil {
+		return fmt.Errorf("Error adding blackhole routes: %v\n", err)
+	}
+	fmt.Printf("Added %d routes.\n", len(addrs))
+	return nil
+}
+
+func UpdateRoutes(ipVersion string, egressFilterList string) error {
+	newAddrs := prepareAddrs(egressFilterList, true)
+	fmt.Printf("Checking ipv%s egress filter list with %d entries against current settings...\n", ipVersion, len(newAddrs))
+
+	currentAddrs, err := GetBlackholeRoutes(ipVersion)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Current filter list contains %d entries\n", len(currentAddrs))
```

Comment:
Please change the log message. This seems to be the number of routes. Hence, you may call it the applied filter list or something similar, but it is not the same as the statement above.

Reply:
Done.